### ISSUE TITLE
[dataclasses] Remove IsPrimary method from I3Particle (Trac #1810)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1802]
This method is useless and often wrong.  There's no good way to ensure it's consistent with the particle's position in an I3MCTree so we should remove it.

Migrated from https://code.icecube.wisc.edu/ticket/1810
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:24", 
    "description": "This method is useless and often wrong.  There's no good way to ensure it's consistent with the particle's position in an I3MCTree so we should remove it.", 
    "reporter": "olivas", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067204154158", 
    "component": "combo core", 
    "summary": "[dataclasses] Remove IsPrimary method from I3Particle", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-08-01T17:00:29", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "cleanup"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20161114T17:44:57: 
* kjmeagher commented:

> In changeset 151491:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="151491"
> remove unused, usless function `I3Particle::IsPrimary()`, fixes #1802
> ```


* kjmeagher changed resolution from "" to "fixed"
* kjmeagher changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:24: 
* olivas changed milestone from "October 2016" to ""
* olivas commented:

> Milestone October 2016 deleted
